### PR TITLE
Add checkbox to the delete worksheet dialog to prevent accidental deletion

### LINKED ISSUE #3312: [FEATURE REQUEST] Add a checkbox before allowing to delete a worksheet
**Is your feature request related to a problem? Please describe.**
Saw in slack (codalab channel) that a user accidentally deleted an entire worksheet instead of bundles, this can be a great cost on the user end to recreate the worksheet.

**Describe the solution you'd like**
Add a checkbox to the delete worksheet dialog and disable the worksheet delete button in the dialog until the checkbox is checked. This would add a barrier and user will have an extra chance to see that it's a worksheet being deleted, not bundle.

**Describe alternatives you've considered**
The freeze/unfreeze feature mentioned in https://github.com/codalab/codalab-worksheets/pull/3295 would be helpful, but still would be nice to have more friction in the worksheet deletion flow, so we reduce accident deletion as much as possible.

### PR DESCRIPTION
**Pull request by yipenghe**
### Reasons for making this change

Adding more friction to the delete worksheet flow on the UI side by adding a checkbox and capitalizing the worksheet term

### Related issues

fix #3312

### Test
Create a new worksheet and use the delete button on top right to delete the worksheet

### Screenshots

Not checked (default when open)
Screenshot from 2021-02-27 14-12-28

Checked:
Screenshot from 2021-02-27 14-12-41

### Checklist

* [x] I've added a screenshot of the changes, if this is a frontend change
* [x] I've added and/or updated tests, if this is a backend change
* [x] I've run the [pre-commit.sh](https://github.com/codalab/codalab-worksheets/blob/master/pre-commit.sh) script
* [x] I've updated [docs](https://github.com/codalab/codalab-worksheets/tree/master/docs), if needed

### CODE REVIEW

**Comment by teetone** [comment]
Putting a checkbox after a colon seems a little odd to me.

What do you think about

```
Delete this worksheet permanently?

[CHECKBOX] Yes, I want to delete this worksheet permanently.
```

**Comment by yipenghe** [comment]
> Putting a checkbox after a colon seems a little odd to me.
> 
> What do you think about
> 
> ```
> Delete this worksheet permanently?
> 
> [CHECKBOX] Yes, I want to delete this worksheet permanently.
> ```

Looks good to me, will update

**Comment by yipenghe** [comment]
> Putting a checkbox after a colon seems a little odd to me.
> 
> What do you think about
> 
> ```
> Delete this worksheet permanently?
> 
> [CHECKBOX] Yes, I want to delete this worksheet permanently.
> ```

I kept the red "cannot be undone" warning, I think this can reduce questions on recovering a deleted worksheet

File: frontend/src/components/worksheets/WorksheetDialogs.js [hunk 1]
```diff
@@ -144,14 +153,24 @@ class WorksheetDialogs extends React.Component {
...
+                            />
+                            {'Yes, I want to delete this worksheet permanently.'}
+                        </DialogContentText>
                         <DialogContentText
                             id='alert-dialog-description'
-                            style={{ color: 'red', marginBottom: '20px' }}
+                            style={{ color: 'red', marginLeft: '35px', marginBottom: '20px' }}
                         >
-                            {'Worksheet deletion cannot be undone.'}
+                            {'WORKSHEET deletion cannot be undone.'}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/WorksheetDialogs.js]:
```suggestion
                            {'Worksheet deletion cannot be undone.'}
```

**Review by epicfaace — APPROVED** [review]

**Pull request merged**